src/components/slide-toggle/slide-toggle.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  Component,
+  ElementRef,
+  Renderer,
+  forwardRef,
+  ChangeDetectionStrategy,
+  Input,
+  Output,
+  EventEmitter
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR
+} from '@angular/common';
+import { BooleanFieldValue } from '../../core/annotations/field-value';
+import { Observable } from 'rxjs/Observable';
+
+export const MD_SLIDE_TOGGLE_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => MdSlideToggle),
+  multi: true
+};
+
+// Increasing integer for generating unique ids for slide-toggle components.
+let nextId = 0;
+
+@Component({
+  selector: 'md-slide-toggle',
+  host: {
+    '[class.md-checked]': 'checked',
+    '[class.md-disabled]': 'disabled',
+  },
+  templateUrl: './components/slide-toggle/slide-toggle.html',
+  styleUrls: ['./components/slide-toggle/slide-toggle.css'],
+  providers: [MD_SLIDE_TOGGLE_VALUE_ACCESSOR],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MdSlideToggle implements ControlValueAccessor {
+
+  private onChange = (_: any) => {};
+  private onTouched = () => {};
+
+  // A unique id for the slide-toggle. By default the id is auto-generated.
+  private _uniqueId = `md-slide-toggle-${++nextId}`;
+  private _checked: boolean = false;
+  private _color: string;
+
+  @Input() @BooleanFieldValue() disabled: boolean = false;
+  @Input() name: string;
+  @Input() id: string = this._uniqueId;
+  @Input() tabIndex: number = 0;
+  @Input() ariaLabel: string = '';
+  @Input() ariaLabelledby: string = null;
+
+  @Output('change') private _change: EventEmitter<boolean> = new EventEmitter<boolean>();
+  change: Observable<boolean> = this._change.asObservable();
+
+  // Returns the unique id for the visual hidden input.
+  getInputId = () => `${this.id || this._uniqueId}-input`;

```

Comment:
@jelbourn Just tested that again, and created some tests, but it seems like we didn't notice, that sometimes the id will set to null. And then we have to fallback to the unique id.

Currently, we will have ids on input like
```
null-input
```